mitmproxy/addons/core.py

```diff
@@ -103,15 +103,15 @@ def resume(self, flows: typing.Sequence[flow.Flow]) -> None:
 
     # FIXME: this will become view.mark later
     @command.command("flow.mark")
-    def mark(self, flows: typing.Sequence[flow.Flow], boolean: bool) -> None:
+    @command.argument("marker", type=mitmproxy.types.Marker)
+    def mark(self, flows: typing.Sequence[flow.Flow], marker: typing.Union[bool, str]) -> None:
         """
             Mark flows.
         """
         updated = []
         for i in flows:
-            if i.marked != boolean:
-                i.marked = boolean
-                updated.append(i)
+            i.marked = marker
```

Comment:
Have you considered doing the `:grape:` -> 🍇 conversion here, i.e. directly apply `render_marker` here?
We could then also fail here if someone specifies an invalid emoji, instead of just silently assuming the default one.

Reply:
I decided not to do this because I assumed that, perhaps, `mitmweb` would render markers differently at some point. Since `mitmweb` isn't showing markers at all (right?) I can leave this abstraction out and just handle it right here as you suggest.

Reply:
Upon further consideration... the other difficulty of rendering the value early is filtering. It is convenient to just search for `~marker circle` and return all flows marked with a circle regardless of the color.

Reply:
I did add a check that raises in that function if the marker is not known. I also supplemented `emoji.py` with the single char `a-zA-Z0-9` values for completeness and to simplify checks elsewhere in the code.